developer-docs-site/static/move_demo/sources/ACLMessageBoard.move

```diff
@@ -0,0 +1,199 @@
+        let post = borrow_global<ACLBasedMB>(board_addr).pinned_post;
+        copy post
+    }
+
+    /// init participant's resource to emit events
+    public fun register_internal(participant: &signer) {
+        move_to(participant, MessageChangeEventHandle{
+            change_events: Event::new_event_handle<MessageChangeEvent>(participant)
+        })
+    }
+
+    public(script) fun register(participant: signer) {
+        register_internal(&participant);
+    }
+
+    /// board owner control adding new participants
+    public fun add_participant_internal(account: &signer, participant: address) acquires ACLBasedMB {
+        let board = borrow_global_mut<ACLBasedMB>(Signer::address_of(account));
+        ACL::add(&mut board.participants, participant);
+    }
+
+    public(script) fun add_participant(account: signer, participant: address) acquires ACLBasedMB {
+        add_participant_internal(&account, participant);
+    }
+
+    /// remove a participant from the ACL
+    public(script) fun remove_participant(account: signer, participant: address) acquires ACLBasedMB {
+        let board = borrow_global_mut<ACLBasedMB>(Signer::address_of(&account));
+        assert!(Signer::address_of(&account) != participant, ECANNOT_REMOVE_ADMIN_FROM_ACL);
+        ACL::remove(&mut board.participants, participant);
+    }
+
+    /// an account publish the message to update the notice
+    public fun send_pinned_message_internal(
+        account: &signer, board_addr: address, message: vector<u8>
+    ) acquires ACLBasedMB, MessageChangeEventHandle {
+        let board = borrow_global<ACLBasedMB>(board_addr);
+        assert!(ACL::contains(&board.participants, Signer::address_of(account)), EACCOUNT_NOT_IN_ACL);
+
+        let board = borrow_global_mut<ACLBasedMB>(board_addr);
+        board.pinned_post = message;
+
+        let send_acct = Signer::address_of(account);
+        let event_handle = borrow_global_mut<MessageChangeEventHandle>(send_acct);
+        Event::emit_event<MessageChangeEvent>(
+            &mut event_handle.change_events,
+            MessageChangeEvent{
+                message,
+                board: board_addr
+            }
+        );
+    }
+
+    public(script) fun send_pinned_message(
+        account: signer, board_addr: address, message: vector<u8>
+    ) acquires ACLBasedMB, MessageChangeEventHandle {
+        send_pinned_message_internal(&account, board_addr, message);
+    }
+
+    /// an account can send events containing message
+    public(script) fun send_message_to(
+        account: signer, board_addr: address, message: vector<u8>
+    ) acquires MessageChangeEventHandle {
+        let send_acct = Signer::address_of(&account);
+        let event_handle = borrow_global_mut<MessageChangeEventHandle>(send_acct);
+        Event::emit_event<MessageChangeEvent>(
+            &mut event_handle.change_events,
+            MessageChangeEvent{
+                message,
+                board: board_addr
```

Comment:
this should be the address of the sender, the events are on the address of the board